include/cpuinfo_x86.h

```diff
@@ -25,6 +25,8 @@ CPU_FEATURES_START_CPP_NAMESPACE
 #define CPU_FEATURES_VENDOR_GENUINE_INTEL "GenuineIntel"
 #define CPU_FEATURES_VENDOR_AUTHENTIC_AMD "AuthenticAMD"
 #define CPU_FEATURES_VENDOR_HYGON_GENUINE "HygonGenuine"
+#define CPU_FEATURES_VENDOR_ZHAOXIN_CTH "CentaurHauls"
+#define CPU_FEATURES_VENDOR_ZHAOXIN_SH "  Shanghai  "
```

Comment:
Could you change `CPU_FEATURES_VENDOR_ZHAOXIN_CTH` and `CPU_FEATURES_VENDOR_ZHAOXIN_SH` to `CPU_FEATURES_VENDOR_CENTAUR_HAULS`, `CPU_FEATURES_VENDOR_SHANGHAI`?

Reply:
My 2cents, I would go for:
```
CPU_FEATURES_VENDOR_CENTAUR
CPU_FEATURES_VENDOR_ZHAOXIN
```
as two separate companies and split code accordingly

src: https://en.wikipedia.org/wiki/Centaur_Technology
src: https://en.wikipedia.org/wiki/Zhaoxin

Reply:
```C
// CPUID Vendors
#define CPU_FEATURES_VENDOR_GENUINE_INTEL "GenuineIntel"
#define CPU_FEATURES_VENDOR_AUTHENTIC_AMD "AuthenticAMD"
#define CPU_FEATURES_VENDOR_HYGON_GENUINE "HygonGenuine"
```
At the moment we specify cpuid vendor string not company, I would like it to be consistent.

There are cases when a company has more than one cpuid vendor string:
`AMDisbetter!`, `AuthenticAMD` - AMD
`TransmetaCPU`, `GenuineTMx86` - Transmeta

Reply:
Both of the VendorID “Shanghai” and "CentaurHauls" belong to Shanghai Zhaoxin Semiconductor Co Ltd. But if the define name only for the vendorID string not company. We think change them to CPU_FEATURES_VENDOR_CENTAUR_HAULS, CPU_FEATURES_VENDOR_SHANGHAI is OK.

Reply:
Yes, I have changed the define name to CPU_FEATURES_VENDOR_CENTAUR_HAULS and CPU_FEATURES_VENDOR_SHANGHAI.